### ISSUE TITLE
DNS record errors

### ISSUE THREAD

**Issue by Axedyson** [vercel/terraform-provider-vercel#62]
Running this terraform code:
```Terraform
terraform {
  required_version = ">= 1.2.7"

  required_providers {
    digitalocean = {
      source  = "digitalocean/digitalocean"
      version = ">= 2.22.1"
    }
    vercel = {
      source  = "vercel/vercel"
      version = ">= 0.7.1"
    }
  }

  cloud {
    organization = "syncbase"

    workspaces {
      name = "syncbase"
    }
  }

resource "vercel_project" "frontend" {
  name      = "frontend-syncbase"
  framework = "nextjs"
}

resource "vercel_project_domain" "default" {
  project_id = vercel_project.frontend.id
  domain     = "syncbase.tv"
}

resource "vercel_dns_record" "a" {
  domain = vercel_project_domain.default.domain
  name   = ""
  type   = "A"
}

resource "vercel_dns_record" "cname" {
  domain = vercel_project_domain.default.domain
  name   = "www"
  type   = "CNAME"
  value  = vercel_project_domain.default.domain
}
```
Raises the following two errors:
```
╷
│ Error: Provider produced inconsistent result after apply
│
│ When applying changes to vercel_dns_record.cname, provider "provider[\"registry.terraform.io/vercel/vercel\"]" produced an unexpected new value: .ttl: was null, but
│ now cty.NumberIntVal(60).
│
│ This is a bug in the provider, which should be reported in the provider's own issue tracker.
╵
╷
│ Error: Error creating DNS Record
│
│   with vercel_dns_record.a,
│   on main.tf line 103, in resource "vercel_dns_record" "a":
│  103: resource "vercel_dns_record" "a" {
│
│ Could not create DNS Record, unexpected error: bad_request - Invalid request: missing required property `value`.
╵
```
As stated in the docs:
<img width="681" alt="image" src="https://user-images.githubusercontent.com/28491857/186758167-2d32bc73-b4e1-4d71-823c-9314a8483294.png">
Both the `ttl` and `value` option are optional so they shouldn't need be specified in the terraform configuration

**Comment by dglsparsons** [comment]
Hey @Axedyson, thanks for raising this. It looks like a bug on our end, I'll get it sorted! :)

**Comment by dglsparsons** [comment]
Hey @Axedyson, some additional context on this since I've been looking with @ecklf :). 

The TTL error is an issue with the provider, we are working on a fix for that at the moment and it should be released shortly. If you need it to work in the meantime you can just specify a `ttl` (60 being the default value). 

For the `value` field, we are looking at improving the documentation and validation, so an `error` should be raised at plan time, rather than apply time. The field is technically optional as far as terraform is concerned - but in reality it is required for most record types. For example, it is not possible to create an `A` record without a `value`, but for `SRV` records the `value` field should not be specified.

**Comment by ecklf** [comment]
Opened up a PR that fixes the issue + documents for which dns record types `value` must be specified. Thanks for reporting the issue.

**Comment by Axedyson** [comment]
@dglsparsons Got thanks!

**Comment by dglsparsons** [comment]
Hey @Axedyson, the `ttl` issue should be fixed as of [v0.8.0](https://github.com/vercel/terraform-provider-vercel/releases/tag/v0.8.0). Could you give it a try and let me know if the issue persists? 

We've also improved the error messages and [documentation](https://registry.terraform.io/providers/vercel/vercel/latest/docs/resources/dns_record) around when the `value` field is required. This should now show an error when you run `terraform plan`, rather than as part of `apply`. Again, please let us know if you encounter anything!

**Comment by Axedyson** [comment]
It works perfectly thanks!
